tools/migrate_kv.ts

```diff
@@ -1,14 +1,140 @@
 // Copyright 2023 the Deno authors. All rights reserved. MIT license.
-import { kv, type Notification } from "@/utils/db.ts";
+import {
+  assertIsEntry,
+  Item,
+  kv,
+  newVoteProps,
+  type Notification,
+  User,
+  Vote,
+} from "@/utils/db.ts";
+
+type OldItem = Omit<Item, "userLogin"> & { userId: string };
+type OldUser = User & { id: string };
+
+async function migrateVote(voterUserId: string, oldItem: OldItem) {
+  const [creatorUserRes, voterUserRes] = await kv.getMany<OldUser[]>([
+    ["users", oldItem.userId],
+    ["users", voterUserId],
+  ]);
+  assertIsEntry(creatorUserRes);
```

Comment:
The distinction between these two types of users must remain correct throughout this function.